src/MentionWrapper.js

```diff
@@ -183,19 +191,29 @@ class MentionWrapper extends Component {
     } = this.props;
     const { active, child, left, top, options } = this.state;
     const { item, className, style } = child;
-    
+
+    const componentProps = {
+      ...inputProps,
+      ref: this.inputRef,
+      onBlur: this.handleBlur,
+      onInput: this.handleInput,
+      onKeyDown: this.handleKeyDown,
+      type: componentInputType || 'input'
+    };
+
+    let inputElement;
+    if (component) {
+      inputElement = React.createElement(component, componentProps);
```

Comment:
I noticed 'component wasnt an added prop.. it was there already... since the beginning.. but i cant tell what it was used for... o.0 instead of using a mystery field, how about we name it InputComponent? Also, do we need to pass in the componentInputType? I think we can just override the given prop in the InputComponent